workers/scanner.go

```diff
@@ -399,6 +425,36 @@ func (s *Scanner) managedRefreshMinPinners() {
 	s.mu.Unlock()
 }
 
+// staticEligibleToPin returns true when this server is either pinning less than
+// AlwaysPinThreshold data or it's in the lower PinningRangeThresholdPercent
+// section of servers when ordered by contract data in descending order.
+// A server is always eligible if it's last in the list.
+func (s *Scanner) staticEligibleToPin(ctx context.Context) (bool, error) {
+	pinnedData, err := s.staticDB.ServerLoad(ctx, s.staticServerName)
+	if err != nil {
+		return false, err
+	}
+	// Below the hard limit.
+	if pinnedData < int64(AlwaysPinThreshold) {
+		return true, nil
+	}
+	pos, total, err := s.staticDB.ServerLoadPosition(ctx, s.staticServerName)
+	if err != nil {
+		return false, err
+	}
+	// Last in the list.
+	if pos == total {
+		return true, nil
+	}
+	// In the bottom PinningRangeThresholdPercent.
+	posPercent := 1.0 - float64(pos)/float64(total)
+	thresholdPercent := float64(PinningRangeThresholdPercent) / 100.0
```

Comment:
nit/tip: you could make the constant `0.3` instead of 30 and dividing by 100.0 to avoid the casting and the floating point division. Doesn't matter too much for this but constants in Go are weird cause they have infinite precision and depending on the value can be assigned to multiple different types without casting. Also more compile time checks. TLDR; constants are nice.

Reply:
> constants are nice.

Yes, very true.

Added. :)